src/gcp_scanner/crawl.py

```diff
@@ -1005,3 +1005,38 @@ def list_services(project_id: str, credentials: Credentials) -> List[Any]:
     logging.info(sys.exc_info())
 
   return list_of_services
+
+
+def list_sourcerepo(project_id: str, credentials: Credentials) -> List[Any]:
+  """Retrieve a list of cloud source repositories enabled in the project.
+
+  Args:
+    project_id: An id of a project to query info about.
+    credentials: An google.oauth2.credentials.Credentials object.
+
+  Returns:
+    A list of cloud source repositories in the project.
+  """
+
+  logging.info("Retrieving cloud source repositories %s", project_id)
+  list_of_repos = list()
+  service = discovery.build("sourcerepo", "v1", credentials=credentials)
+
+  request = service.projects().repos().list(
+    name="projects/" + project_id,
+    pageSize=200
```

Comment:
Is 500 maximum? Let's try to fetch as much data as possible with one query.

Reply:
Agreed, but I think this line handles the paginated response. So, if there are more repos, eventually that will be fetched. 
I took the example from `list_services` function which was previously implemented. My understanding could be wrong in this case, looking froward to your opinion.

Reply:
Yes, I know that eventually everything will be fetched, we just want to reduce overall amount ofl interaction (requests) with Cloud API.

Reply:
great, understood clearly.